### ISSUE TITLE
Combination crafting does not work with very high energy values

### ISSUE THREAD

**Issue by DerCommander323** [OmnifactoryDevs/ExtendedCrafting#21]
This is a bug I've encountered on the normal mod version aswell, but if an item made with combination crafting requires a very large amount of energy in total to craft (i assume the limit is 2.147 billion), the item will never finish crafting, it will just forever keep using energy. Using Integrated Dynamics to read the nbt of the crafting core quickly, it is possible to see that the 'Progress' value constantly switches between negative and positive numbers, which I assume is because of the integer limit.

**Comment by eutro** [comment]
Can you post an example of a script that adds such a recipe, and describe the setup: for example, how are you supplying power? I'm wondering if this is just an overflow as you transfer too much power at once.

**Comment by eutro** [comment]
Yeah, no, looks like it's just that recipes support cost up to long limit

https://github.com/OmnifactoryDevs/ExtendedCrafting/blob/7a1f8c36853880ea9eaebfbebb4b094eec0eef4e/src/main/java/com/blakebr0/extendedcrafting/crafting/CombinationRecipe.java#L16

but the tile is only storing its own power as int

https://github.com/OmnifactoryDevs/ExtendedCrafting/blob/7a1f8c36853880ea9eaebfbebb4b094eec0eef4e/src/main/java/com/blakebr0/extendedcrafting/tile/TileCraftingCore.java#L40

[In Cucumber](https://github.com/BlakeBr0/Cucumber/blob/2f5a15559db163c0ab5aa4cfb81bec3bacf2c50d/src/main/java/com/blakebr0/cucumber/energy/EnergyStorageCustom.java#L7)

[In Forge](https://github.com/MinecraftForge/MinecraftForge/blob/d3f01843f7e7a4f613b5e8113d381fd8747b4343/src/main/java/net/minecraftforge/energy/EnergyStorage.java#L30)

**Comment by DerCommander323** [comment]
wouldnt it just be that this variable has to be changed to a long?
https://github.com/OmnifactoryDevs/ExtendedCrafting/blob/7a1f8c36853880ea9eaebfbebb4b094eec0eef4e/src/main/java/com/blakebr0/extendedcrafting/tile/TileCraftingCore.java#L42

since further down at
https://github.com/OmnifactoryDevs/ExtendedCrafting/blob/7a1f8c36853880ea9eaebfbebb4b094eec0eef4e/src/main/java/com/blakebr0/extendedcrafting/tile/TileCraftingCore.java#L147
from what im understanding its just subtracting the progress value which increases every tick by the energy that the recipe consumed from the total recipe energy cost, until it reaches 0, but since progress isnt a long it will overflow at over 2b and thus if the total energy cost is over that limit it can never subtract enough to bring it to 0 and finish, and if it was a long it could do that, right?
